### DISCUSSION TITLE
An error occurred while executing 'GetAll'&ToList in the service

### DISCUSSION THREAD

**Discussion by hehetiger** [abpframework/abp#21063] [Q&A]
private readonly IRepository<MesFactory, int> _factoryRepository;

public FactoryModelsAppService(
    IRepository<MesFactory, int> factoryRepository)
{
    _factoryRepository = factoryRepository;
}

public async Task<ListResultDto<FactoryModelDto>> GetFactoryModels()
{
    var query = _factoryRepository.GetAll().ToList();

    var dtos = ObjectMapper.Map<List<MesFactoryDto>>(query);

    if (dtos != null && dtos.Any())
    {
        foreach (var dto in dtos)
        {
            dto.Level = 1;
        }
    }
    return new ListResultDto<FactoryModelDto>(dtos.OrderBy(x => x.Name).ToList());
}

The database, entity, and DTO are all of type int and do not have fields of type long. However, when executing 'GetAll'.'ToList', an exception 'Unable to cast object of type' System 'is thrown Int32' to type 'System.Int64''
![企业微信截图_17287020872102](https://github.com/user-attachments/assets/29b7e36b-6e70-4bbf-8d2b-4305b05780bf)
![企业微信截图_17287021082641](https://github.com/user-attachments/assets/132f5cdf-2b85-4024-ab22-39d19f352094)
![企业微信截图_17287021182217](https://github.com/user-attachments/assets/f64d04bf-8841-4a25-9ab1-392b60d44354)
![企业微信截图_17287021372535](https://github.com/user-attachments/assets/b12bbf5a-3cbc-4b1e-94d9-1677a6a00504)

**Comment by bsogulcan** [comment]
Hi, is your db migrations are up to date? and i wonder 'repository.GetAll().ToList()' is normally working?
Can you try GetListAsync in [docs](https://abp.io/docs/latest/framework/architecture/domain-driven-design/repositories#standard-repository-methods)

**Reply by hehetiger** [reply]
There is no function GetListAsync , just GetAllListAsync. And the GetAllListAsync throw the same exception
What surprises me is , 'User','Role','Organization' is normally working with '.GetAll().ToList()'  ,I don't know why this new module is wrong even the code is same
